src/ota/mod.rs

```diff
@@ -104,3 +126,2109 @@ impl Default for DeployStatus {
+            OtaStatus::Deployed(req) => write!(f, "Deployed {req}"),
+            OtaStatus::Rebooting(req) => write!(f, "Rebooting {req}"),
+            OtaStatus::Rebooted => write!(f, "Rebooted"),
+            OtaStatus::Success(req) => write!(f, "Success {req}"),
+            OtaStatus::Error(err, req) => write!(f, "Error {req}: {err}"),
+            OtaStatus::Failure(err, req) => {
+                write!(f, "Failure")?;
+
+                if let Some(req) = req {
+                    write!(f, " {req}")?;
+                }
+
+                write!(f, ": {err}")
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OtaId {
+    pub uuid: Uuid,
+    pub url: String,
+}
+
+impl From<OtaRequest> for OtaId {
+    fn from(value: OtaRequest) -> Self {
+        Self {
+            uuid: value.uuid.0,
+            url: value.url,
+        }
+    }
+}
+
+impl Display for OtaId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.uuid)
+    }
+}
+
+/// Provides OTA resource accessibility only by talking with it.
+pub struct Ota<T, U>
+where
+    T: SystemUpdate,
+    U: StateRepository<PersistentState>,
+{
+    pub system_update: T,
+    pub state_repository: U,
+    pub download_file_path: PathBuf,
+    pub ota_status: OtaStatus,
+    pub flag: OtaInProgress,
+    pub publisher_tx: mpsc::Sender<OtaStatus>,
+}
+
+#[async_trait]
+impl<T, U> Actor for Ota<T, U>
+where
+    T: SystemUpdate,
+    U: StateRepository<PersistentState>,
+{
+    type Msg = OtaMessage;
+
+    fn task() -> &'static str {
+        "ota"
+    }
+
+    async fn init(&mut self) -> stable_eyre::Result<()> {
+        if self.state_repository.exists().await {
+            self.ota_status = OtaStatus::Rebooted;
+        }
+
+        // Not cancellable after a reboot, this is just a placeholder
+        let cancel = CancellationToken::new();
+
+        // Always run the publish and cleanup
+        self.handle_ota_update(cancel).await;
+
+        Ok(())
+    }
+
+    async fn handle(&mut self, msg: Self::Msg) -> stable_eyre::Result<()> {
+        if self.ota_status != OtaStatus::Idle {
+            error!("ota request already in progress");
+
+            return Err(OtaError::UpdateAlreadyInProgress.into());
+        }
+
+        self.ota_status = OtaStatus::Init(msg.ota_id);
+
+        self.handle_ota_update(msg.cancel).await;
+
+        Ok(())
+    }
+}
+
+impl<T, U> Ota<T, U>
+where
+    T: SystemUpdate,
+    U: StateRepository<PersistentState>,
+{
+    pub fn new(
+        opts: &DeviceManagerOptions,
+        tx_publisher: mpsc::Sender<OtaStatus>,
+        flag: OtaInProgress,
+        system_update: T,
+        state_repository: U,
+    ) -> Result<Self, DeviceManagerError> {
+        Ok(Ota {
+            system_update,
+            state_repository,
+            download_file_path: opts.download_directory.clone(),
+            ota_status: OtaStatus::Rebooted,
```

Comment:
Why is this still in Rebooted, and not in Idle?

Reply:
My mistake, now it's fixed